src/index.ts

```diff
@@ -1,33 +1,31 @@
-import { Users } from './services/users';
-import { Logger } from './services/logger';
+import { createIoCContainer } from "./ioc";
+import type { User } from "./types";
 
-import type { User, ApiConfig } from './types';
+const config = (window as any).__CONFIG__;
+delete (window as any).__CONFIG__;
 
-const renderUsers = async (config: ApiConfig) => {
-  const usersService = new Users(config);
+const ioc = createIoCContainer();
+ioc.register("config", config.api);
+
+const renderUsers = async (ioc: any) => {
```

Comment:
there is no reason to take ioc as an argument, because it's available within scope (line 7)